test/e2e/antctl_test.go

```diff
@@ -0,0 +1,114 @@
+package e2e
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func antctlOutput(stdout, stderr string, tb testing.TB) {
+	tb.Logf("antctl stdout:\n%s", stdout)
+	tb.Logf("antctl stderr:\n%s", stderr)
+}
+
+func runAntctl(podName string, subCmds []string, data *TestData, tb testing.TB) (string, string, error) {
+	var containerName string
+	if strings.Contains(podName, "agent") { // simple hack here
+		containerName = "antrea-agent"
+	} else {
+		containerName = "antrea-controller"
+	}
+
+	cmds := []string{"antctl", "-v"}
+	stdout, stderr, err := data.runCommandFromPod(AntreaNamespace, podName, containerName, append(cmds, subCmds...))
+	antctlOutput(stdout, stderr, tb)
+
+	return stdout, stderr, err
+}
+
+func TestAntctlAgentLocalAccess(t *testing.T) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+	podName, err := data.getAntreaPodOnNode(masterNodeName())
+	if err != nil {
+		t.Fatalf("Error when getting antrea-agent pod name: %v", err)
+	}
+	if _, _, err := runAntctl(podName, []string{"version"}, data, t); err != nil {
+		t.Fatalf("Error when running `antctl version` from %s: %v", podName, err)
+	}
+}
+
+func TestAntctlControllerLocalAccess(t *testing.T) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+	podName, err := data.getAntreaController()
+	if err != nil {
+		t.Fatalf("Error when getting antrea-controller pod name: %v", err)
+	}
+	if _, _, err := runAntctl(podName, []string{"version"}, data, t); err != nil {
+		t.Fatalf("Error when running `antctl version` from %s: %v", podName, err)
+	}
+}
+
+func TestAntctlControllerRemoteAccess(t *testing.T) {
+	data, err := setupTest(t)
+	if err != nil {
+		t.Fatalf("Error when setting up test: %v", err)
+	}
+	defer teardownTest(t, data)
+
+	podName, err := data.getAntreaController()
+	assert.Nil(t, err, "Error when retrieving antrea controller pod name")
+
+	cmd := fmt.Sprintf("kubectl cp %s/%s:/usr/local/bin/antctl ~/antctl", AntreaNamespace, podName)
+	_, _, _, err = RunCommandOnNode(masterNodeName(), cmd)
+	assert.Nil(t, err, "Error when copying antctl from %s", podName)
+
+	_, stdout, stderr, err := RunCommandOnNode(masterNodeName(), "~/antctl -v version")
```

Comment:
The `antctl` is copied from pod. From the stderr and stdout, I think this test runs as I expected.
But you are right, I will check return code in future.
Just added some return code assertions.

Reply:
Thanks, I completely missed that the `antclt` binary was built by `make bin`, so I was very confused by what was going on.

That makes the `antctl` target confusing to the user though, because there is a target (`bin`) that produces a binary with this name but it's just not that one. For a given OS `X`, we also end up with 2 identical binaries with different names: `antrea-X` and `antrea`. Documentation may help there, but any thoughts on making this more consistent?

Reply:
In my thought, the `antctl-X` is mainly for releases, but I don't have a good idea about where to put them.

Reply:
Use #312 to track this issue.